### ISSUE TITLE
cant edit markdown files

### ISSUE THREAD

**Issue by mstyp** [nextcloud/android#9771]
### Steps to reproduce
1. open app.
2. click 3 dot menue next to share button next to file name
3. click edit
4. spinning blue wheel appears
5. after a while little grey box appears saying "loading takes very long..." but no other info
6. wait. wait. wait. nothing ever happens.

### Expected behavior
Honestly I dont know what should happen. I have never actually seen the file editer. Its never worked through all the various versions of the app. 

### Actual behaviour
Just loads forever. 

### Can you reproduce this problem on https://try.nextcloud.com?
No. This tablet **by design** does not connect to the internet. It only ever connects to a local lan, and the server only ever runs on the local lan with no connection to the broader internet (other then me moving files over by USB stick). so no cant reproduce on random internet facing service. 

### Environment data
Android version: 10

Device model: Samsung tablet

Stock or customized system: NA

Nextcloud app version: Not sure, still waiting to see if the app times out or gives some other information or something. Will wait five minutes then edit this comment. 

Nextcloud server version:  22.2.3

Reverse proxy: idk what that is

### Logs
its really hard to annonimise logs so I wont bother unless somone tells me its absolutly nessisary.

**Comment by yamsu** [comment]
I have had this issue when my ssl certificate was not secure/incorrectly installed. Once the ssl certificate was corrected the editor worked without trouble.

**Comment by mstyp** [comment]
> I have had this issue when my ssl certificate was not secure/incorrectly installed. Once the ssl certificate was corrected the editor worked without trouble.

do you mean I cant use a self signed cert or do you mean I need to do some other fix? My server never touches the internet and uses nextcloud.local as the domain so I'mnot sure I can get anything *but* a self signed cert.

**Comment by AlvaroBrey** [comment]
Hm, this is possibly actually a result of the network being airgapped. The file editor is served from the Nextcloud server as a webpage, however it _should_ be working as all assets should be served directly from Nextcloud without internet access.

Do you have an up-to-date Webview implementation installed in your tablet?

**Comment by AlvaroBrey** [comment]
> According to the settings I have enabled:
> 
> Android System WebView 80.0.3987.99
> 
> Which does look out of date, but I'm not sure If I can updated it. Is that version unsupported?

No idea, we don't keep track of supported WebView versions AFAIK.

Another question: Does the file editor work from the web ui? (just login into the web UI and click a text file)

**Comment by mstyp** [comment]
> Another question: Does the file editor work from the web ui? (just login into the web UI and click a text file)

Yes, markdown webui works fantastic on desktop Firefox and extraordinarily slowly on android Firefox (well once it loads its fine just takes ~10 seconds to load). chrome browser is a bit faster on android. 

I should also note that, if I have downloaded the file to the tablet, I can *read* markdown and txt files just fine on the android app. its only when I try to edit them that an infanate loop starts.

**Comment by mstyp** [comment]
I attempted to check the logs for my nextcloud server to see if they revealed anything, but there is absolutely nothing different between the logs for when I try to edit a file, and just not doing anything at all for 30 minutes. 

Is there a way to check error logs just for the nextcloud android app?

**Comment by mstyp** [comment]
It seems that I *can* edit markdown files with the markdown dev version. So now I'm not sure whats up.

**Comment by AlvaroBrey** [comment]
> Is there a way to check error logs just for the nextcloud android app?
https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat-mag explains how to using a terminal

**Comment by AlvaroBrey** [comment]
Can any of the affected users try the QA apk in https://github.com/nextcloud/android/pull/9881#issuecomment-1050070378?

**Comment by mstyp** [comment]
> Can any of the affected users try the QA apk in https://github.com/nextcloud/android/pull/9881#issuecomment-1050070378?

Hey, apologies for delay following up on this but yes this QA apk does allow me to edit text files! Awesome! 

I looked at that discussion thread and I think you might have already decided to implement it, and if so I'm excited!
